### PR TITLE
Convert README to Markdown

### PR DESCRIPTION
**Pull request by chrysle**
Part of #1845, picks up work from and closes #1862 

##### Contributor checklist

- [ ] Provided the tests for the changes.
- [x] Assure PR title is short, clear, and good to be included in the user-oriented changelog

##### Maintainer checklist

- [x] Assure one of these labels is present: `backwards incompatible`, `feature`, `enhancement`, `deprecation`, `bug`, `dependency`, `docs` or `skip-changelog` as they determine changelog listing.
- [ ] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: README.md
Comment (chrysle) [inline README.md:47]: I think I'm going to make a relative link out of that.

File: pyproject.toml
Comment (webknjaz) [inline pyproject.toml:11]: I wonder if this will need a long description type set explicitly...

Comment (webknjaz) [reply pyproject.toml:11]: Try building the dist via `python -m dist` (there might be a tox env for this) and inspect it with `twine --strict`.

File: docs/index.md
Comment (webknjaz) [inline docs/index.md:1]: I'd probably put the Sphinx stuff into a separate PR if possible.

File: docs/conf.py
Comment (webknjaz) [inline docs/conf.py:55]: Is it possible not to ignore this? (also a separate PR candidate)

File: README.md
Comment (webknjaz) [inline README.md:395]: YAML indentation is broken here, let's try this:
```suggestion
- repo: https://github.com/jazzband/pip-tools
  rev: 6.13.0
  hooks:
  - id: pip-compile
```

File: README.md
Comment (webknjaz) [inline README.md:402]: Same here

File: README.md
Comment (webknjaz) [inline README.md:414]: and here

File: README.md
Comment (webknjaz) [inline README.md:501]: Plz move this to a detached reference.

File: README.md
Comment (webknjaz) [inline README.md:431]: Make sure to incorporate https://github.com/jazzband/pip-tools/pull/1847 and other commits that changed `README.rst` post #1862: https://github.com/jazzband/pip-tools/commits/main/README.rst.

File: pyproject.toml
Comment (chrysle) [reply pyproject.toml:11]: > I wonder if this will need a long description type set explicitly...

I changed this line because I received a traceback about missing long description.

File: docs/conf.py
Comment (chrysle) [reply docs/conf.py:55]: It'd be needed to create [cross-reference targets](https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html) for all the detached links and images.

File: docs/index.md
Comment (chrysle) [reply docs/index.md:1]: This would break the tests?

File: README.md
Comment (chrysle) [reply README.md]: `pre-commit` caught the indentation for us ;-)

File: pyproject.toml
Comment (webknjaz) [reply pyproject.toml:11]: I know. I'm just not sure if that should be paired with another setting.

File: docs/index.md
Comment (webknjaz) [reply docs/index.md:1]: Not necessarily. You could include the new file instead of rewriting an extra document entirely…

File: docs/conf.py
Comment (webknjaz) [reply docs/conf.py:55]: How so?

File: pyproject.toml
Comment (webknjaz) [reply pyproject.toml:11]: Look at the table @ https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata:
It says:
> If `content-type` is not given, `"text/x-rst"` is used by default.

So I'm almost sure this would render poorly on PyPI. Could you build the package off of this branch, then unpack the tarball (`python -m tarfile`) and the wheel (`wheel unpack`) and look into the metadata files to see if there's a mention of the markdown type? If it doesn't say markdown, maybe we have to set that explicitly.

File: docs/index.md
Comment (webknjaz) [inline docs/index.md:1]: This could also be submitted separately and just include the RST file.

File: pyproject.toml
Comment (chrysle) [reply pyproject.toml:11]: The METADATA file of the wheel and the PKG-INFO of the built package contain the following line:

```
Description-Content-Type: text/markdown
```

This seems to point out that explicitly specifying the content type in the pyproject.toml is not necessary. I would also be surprised about everything else, since the [python packaging guide](https://packaging.python.org/en/latest/tutorials/packaging-projects/#configuring-metadata) recommends this configuration.

File: docs/conf.py
Comment (chrysle) [reply docs/conf.py]: It's fixed here, but there will be some extra work once I convert the index in the other PR. Let's merge this first?

File: docs/index.md
Comment (chrysle) [reply docs/index.md:1]: I've moved the conversion onto another branch. But I'd wait until this is merged and submit the PR including the newly converted Markdown README.

File: pyproject.toml
Comment (webknjaz) [reply pyproject.toml:11]: Marking the thread as resolved.

File: docs/index.rst
Comment (chrysle) [reply docs/index.rst:11]: Apparently not... I tried that out, but it didn't work even when I adjusted the [configuration](https://www.sphinx-doc.org/en/master/usage/markdown.html#configuration).

> Also, did you compare the rendering of the current RTD front page vs. this PR?

They differ a little, the size of the captions is not exactly the same, and the new version has a table of contents for each page. But I think it's all right (you can also check the build result yourself one last time).

Comment (webknjaz) [reply docs/index.rst:11]: Ah, maybe that setting doesn't apply to includes..

Comment (webknjaz) [reply docs/index.rst:11]: As for the title sizes, I think MyST has an option to offset that.

Comment (chrysle) [reply docs/index.rst:11]: There are some more problems, as I realized while scanning through the RTD build.

Regarding the badges and picture at the beginning of the README, the reference-style links don't render correctly.
Furthermore, syntax and console highlighting seem broken in some places. Also, the version table a the end is broken.

I do not know why all this renders correctly locally, but not here.

File: README.md
Comment (webknjaz) [inline README.md:1]: I'm almost sure this should be something like
```suggestion
[![jazzband-image]][jazzband]
```

File: docs/index.rst
Comment (webknjaz) [reply docs/index.rst:11]: Added a comment on the relevant line.

Comment (webknjaz) [reply docs/index.rst:11]: > Also, the version table at the end is broken.

That would have to be fixed before merging. Tables can be tricky (both in MD and RST). I'd start by copying a table from elsewhere and adopting it. You could also try removing some lines to see if that makes it work — this would mean that only some lines might be broken and not the entire thing.

File: README.md
Comment (webknjaz) [inline README.md:561]: Could the `*` be breaking this table?

File: README.md
Comment (webknjaz) [inline README.md:20]: Oh, I see that this isn't highlighted at all.. I wonder if it's a MyST bug and should be reported to @chrisjsewell. Also, could it be because of the inclusion of MD-into-RST and not MD-into-MD? I wonder if we converted `docs/index.rst` to `docs/index.md` first, would it break the same way (with `README.rst` remaining the same)?

File: README.md
Comment (chrysle) [reply README.md:1]: But what's about the syntax for the [alt text](https://www.w3schools.io/file/markdown-images/#how-to-add-an-image-with-a-link)?

File: README.md
Comment (chrysle) [reply README.md:561]: But it's escaped?

File: README.md
Comment (chrysle) [reply README.md:20]: Yes, #1874 seems broken....

File: README.md
Comment (webknjaz) [reply README.md:561]: Yes, though I'm not sure Markdown has a concept of escaping like RST. So it might be parser-dependent. GH uses GFM and Sphinx has MyST integrated. This is why more experimentation may be needed to make sure it works universally.

File: README.md
Comment (webknjaz) [reply README.md:1]: The alt text is what's in the first pair of brackets, the second one is the link to the thing. If you want different alt text, you could change the link definition, I think. Also, Markdown has various extensions that may or may not be supported by different MD parsers. I'm pretty sure MyST is mostly bare-bones, except for its own addition for Sphinx/RST compat...

**Review by ssbarnea — CHANGES_REQUESTED** [review]
It looks almost ready but that tables is borked. I would find acceptable to get rid of the table and replace it with something like a simple list. Another alternative would be to make the table plain text and escape it as a code block.

In general I am for avoiding tables as they do not render well on mobile devices. When possible to avoid tables, I usually do.

**Comment by chrysle** [comment]
Everything looks good now!

**Review by webknjaz — APPROVED** [review]

**Comment by chrysle** [comment]
@ssbarnea Could you take another look?

**Review by ssbarnea — APPROVED** [review]

**Pull request merged**
